src/utils/comparePurchaseUrgency.js

```diff
@@ -0,0 +1,49 @@
+import { Timestamp } from 'firebase/firestore';
+import { getDaysBetweenDates, itemIsOverdue, itemIsInactive } from './dates';
+
+//comparison function to be passed into Array.sort()
+export function comparePurchaseUrgency(a, b) {
+	/*
+	Here we are prioritizing sorting inactive and/or overdue items first.
+	Overdue items are given priority and placed highest on the list.
+	Inactive items are given lower priority and placed lowest on the list.
+	*/
+	if (!itemIsInactive(a) && itemIsInactive(b)) {
+		return -1;
+	} else if (itemIsInactive(a) && !itemIsInactive(b)) {
+		return 1;
+	} else if (itemIsInactive(a) && itemIsInactive(b)) {
+		return a.name.localeCompare(b.name);
+	}
+
+	if (itemIsOverdue(a) && !itemIsOverdue(b)) {
+		return -1;
+	} else if (!itemIsOverdue(a) && itemIsOverdue(b)) {
+		return 1;
+	} else if (itemIsOverdue(a) && itemIsOverdue(b)) {
+		//if both items are overdue, sort alphabetically
+		return a.name.localeCompare(b.name);
+	}
+
```

Comment:
I think we can improve the readability of this code a bit down the line. For example, by sorting of daysUntilNext purchase if you were to utilize the firebase function to return the list pre-sorted alphabetically. But for now, it works and it meets the acceptance criteria :)